### PR TITLE
pass real domain to get_test_payload

### PR DESCRIPTION
**Pull request by sravfeyn**
@snopoke  http://manage.dimagi.com/default.asp?231553

### CODE REVIEW

File: corehq/apps/repeaters/repeater_generators.py [hunk 1]
```diff
@@ -55,8 +55,8 @@ class FormRepeaterXMLPayloadGenerator(BasePayloadGenerator):
     def get_payload(self, repeat_record, payload_doc):
         return payload_doc.get_xml()
 
-    def get_test_payload(self):
-        return self.get_payload(None, _get_test_form())
+    def get_test_payload(self, domain=None):
+        return self.get_payload(None, _get_test_form(domain))
```
Comment (sravfeyn) [inline corehq/apps/repeaters/repeater_generators.py]: Note that here `self.repeater.domain` is not set, so the domain has to be passed all the from view

File: corehq/apps/repeaters/repeater_generators.py [hunk 2]
```diff
@@ -40,7 +40,7 @@ def get_payload(self, repeat_record, payload_doc):
     def get_headers(self):
         return {'Content-type': self.content_type}
 
-    def get_test_payload(self):
+    def get_test_payload(self, domain=None):
```
Comment (snopoke) [inline corehq/apps/repeaters/repeater_generators.py]: Curious why you made it optional.

**Comment by snopoke** [comment]
:+1: minor comment

File: corehq/apps/repeaters/repeater_generators.py (hunk 1 above)
Comment (sravfeyn) [inline corehq/apps/repeaters/repeater_generators.py]: Most of the generators don't really need it, only two of them need it

File: corehq/apps/repeaters/repeater_generators.py (hunk 2 above)
Comment (sravfeyn) [inline corehq/apps/repeaters/repeater_generators.py]: Not all generators need it, only 2 of them require it

Comment (snopoke) [inline corehq/apps/repeaters/repeater_generators.py]: Sure but if even one requires it then they should all require it to be safe surely?

Comment (sravfeyn) [inline corehq/apps/repeaters/repeater_generators.py]: I don't follow 'to be safe' but I thought it's not ideal to force callers to pass an argument that doesn't need to be get passed. (changed it anyway)

**Comment by sravfeyn** [comment]
@snopoke

**Pull request merged**